core/vm/instructions.go

```diff
@@ -667,6 +667,24 @@ func opJumpi(pc *uint64, interpreter *EVMInterpreter, contract *Contract, memory
 }
 
 func opJumpdest(pc *uint64, interpreter *EVMInterpreter, contract *Contract, memory *Memory, stack *Stack) ([]byte, error) {
+        return nil, nil
+}
+
+func opJumpSub(pc *uint64, interpreter *EVMInterpreter, contract *Contract, memory *Memory, stack *Stack, returns *Stack) ([]byte, error) {
+	pos := stack.pop()
+	returns.push(pos)
+	if !contract.validJumpdest(pos) {
```

Comment:
wouldn't it make more sense to validate before pushing to `returns` ?

Reply:
I'm sorry, I haven't set up the stack yet.  The trick is to put the code size as the first entry.  So popping that item will cause a jump into the implicit zeros at the end of the code and stop.

Reply:
Whoops, wrong panic.  The trick above applies to the returns stack, so there is no danger of panic there.

Reply:
Here, yes, check first.